cmd/controller/main.go

```diff
@@ -68,22 +88,80 @@ func main() {
 	flag.Parse()
 	logf.SetLogger(logf.ZapLogger(false))
 
-	// Read the logging config and setup a logger.
-	cm := getLoggingConfigOrDie()
-
-	config, err := logging.NewConfigFromMap(cm, logconfig.Controller)
-	if err != nil {
-		log.Fatalf("Error parsing logging configuration: %v", err)
-	}
-	logger, atomicLevel := logging.NewLoggerFromConfig(config, logconfig.Controller)
+	logger, atomicLevel := setupLogger()
 	defer logger.Sync()
 	logger = logger.With(zap.String(logkey.ControllerType, logconfig.Controller))
 
-	logger.Info("Starting the controller")
-
 	// set up signals so we handle the first shutdown signal gracefully
 	stopCh := signals.SetupSignalHandler()
 
+	go startPkgController(stopCh, logger, atomicLevel)
+	go startControllerRuntime(stopCh, logger, atomicLevel)
+}
+
+func startPkgController(stopCh <-chan struct{}, logger *zap.SugaredLogger, atomicLevel zap.AtomicLevel) {
+	logger = logger.With(zap.String("controller/impl", "pkg"))
+	logger.Info("Starting the controller")
+
+	cfg, err := clientcmd.BuildConfigFromFlags(*masterURL, *kubeconfig)
+	if err != nil {
+		logger.Fatalw("Error building kubeconfig", zap.Error(err))
+	}
+
+	const numControllers = 1
+	cfg.QPS = numControllers * rest.DefaultQPS
+	cfg.Burst = numControllers * rest.DefaultBurst
+	opt := reconciler.NewOptionsOrDie(cfg, logger, stopCh)
+
+	kubeInformerFactory := kubeinformers.NewSharedInformerFactory(opt.KubeClientSet, opt.ResyncPeriod)
+	eventingInformerFactory := informers.NewSharedInformerFactory(opt.EventingClientSet, opt.ResyncPeriod)
+
+	subscriptionInformer := eventingInformerFactory.Eventing().V1alpha1().Subscriptions()
+	//deploymentInformer := kubeInformerFactory.Apps().V1().Deployments()
```

Comment:
remove?

Reply:
I want to keep them until I am finished with the the other rec loops.